src/main/java/com/redhat/rhjmc/containerjfr/net/web/WebServer.java

```diff
@@ -534,6 +542,25 @@ void handleAuthRequest(RoutingContext ctx) {
         }
     }
 
+    void handleHealthRequest(RoutingContext ctx) {
+        ctx.response().putHeader(HttpHeaders.CONTENT_TYPE, MIME_TYPE_JSON);
+        ctx.response().setStatusCode(200);
+
+        boolean datasourceAvailable = false;
+
+        if (this.env.hasEnv(GRAFANA_DATASOURCE_ENV)) {
+            HttpGet grafanaHealth = new HttpGet(this.env.getEnv(GRAFANA_DATASOURCE_ENV));
+            try {
+                CloseableHttpResponse response = this.client.execute(grafanaHealth);
```

Comment:
I think the `response` needs to be cleaned up - simply `try (CloseableHttpResponse response = ...) { ... }` would suffice.